### DISCUSSION TITLE
ご挨拶

### DISCUSSION THREAD

**Discussion by tsukada-shigeki** [toppers/users-forum#68] [TOPPERS Seminar]
はじめまして
塚田と申します。宜しくお願いします。

Toppersセミナーを受講しています。
新基礎３篇：１日目で質問が有ります。
１）添付写真は昨年のtoppers資料を用いて実行したlcd_initでの結果です。
lcdドライバはドライバはadafruit_st7735です。
2)教育セミナーの視聴ページ；新基礎３編１日目117ページによればasp\gdic直下に[adafruit_st7789]フォルダが有りますが、現状PCにはadafruit_st7735のみあります。
再度ASP/プラットフォーム環境構築を行えばよいでしょうか？ご教示願います。

![LCD_INIT](https://user-images.githubusercontent.com/106538497/171008782-0472fdb4-0ced-4fcb-b8fe-d5aebb4210c0.jpg)
![LCDドライバの設定](https://user-images.githubusercontent.com/106538497/171008807-01237e2f-675a-40e3-94e8-eae761b8776f.png)

**Comment by Ryohsuke-Takeuchi** [comment]
adafruit_st7789ドライバは、TOPERS BASE PLATFORM(ST)中にあります。
対応はV1.4.1からです。
現状の塚田様のgdicディレクトリを参照できないので、なぜ、adafruit_st7789がないのかよくわかりませんが、
V1.4.1より古いものを使用しているのでは、ないかと思います。
SDカードのplatform/asp_baseplatformv1.4.2_052421.tar.gzをaspディレクトリと並列の場所において
tar zxvf asp_baseplatformv1.4.2_052421.tar.gz(return)で、再設定するか、
または、最新のasp_baseplatformv1.4.3_042422.tar.gzをTOPPERSのWebからダウンロードして
同等に、tarコマンドで再設定すれば、gdic/adafruit_st7789は作成していると思います。

**Reply by tsukada-shigeki** [reply]
竹内様
お世話になります。
ご回答拝受致しました。有難うございます。
ご指摘の通りV1.4.1より古いV1.3.0を使っていたことが原因でした。本日最新の物をダウンロードし、問題は解決しました。有難うございました。
塚田

**Comment by Ryohsuke-Takeuchi** [comment]
塚田様
arm-none-eabi-gccに関しても、TOPPERSのリアルタイムカーネルでは、V4.6.3やV4.9等の古いものを推奨しています。公開されているarm-none-eabi-gccの最新版はV10.2.1等ですが、リアルタイムカーネルでは、問題が発生する可能性があります。今回、教材で使っているasp-1.9.3とasp_arch_arm_m3_gcc-1.9.8の組み合わせでは、arm-none-eabi-gccV10.2.1上でも動作することを確認しています。できるだけ、新しいものの組み合わせで学習を行った方が問題が発生しないと思います。よろしくお願いいたします。
